ocaml/typing/typedecl.ml

```diff
@@ -2217,11 +2218,33 @@ let error_if_has_deep_native_repr_attributes core_type =
   in
   default_iterator.typ this_iterator core_type
 
-let make_native_repr env core_type sort ty ~global_repr =
+(* Note that [typ] is guaranteed not to contain sort variables because it was
+   produced by [type_scheme], which defaults them.
+
+   However, there can be jkind [any] present with something like:
+    [external f : ('a : any). 'a -> 'a = "%identity"]
+   The current [type_sort_external] implementation will force ['a] to jkind [value]
+   and not produce an error. It will get typed as:
+    [external f : 'a -> 'a = "%identity"]
+
+   Should consider revisiting this decision. *)
```

Comment:
I think my above comment is slightly confused, in particular the question

> Are there any builtins that work on a genuine (non-rep-poly) any?

Is irrelevant, because you can't actually get an any, as noted earlier in the comment you get a sort variable.  So I believe there is clearly no use for this and we should just ban it while we're here.

Reply:
What about `raise : ('a : any). exception -> 'a`?

Reply:
The point is that with the current implementation, if you write:

```ocaml
external raise : ('a : any). exception -> 'a
```

You get a sort variable that is eventually defaulted to value if unused - it's not actually layout polymorphic.  For layout polymorphism you have to add the attribute.  And we can easily make a list of the primitives that work with the attribute.

(Separately, raise already has this type, it doesn't need to be an external, so I'm not sure it's a compelling example.)

Reply:
OK

Reply:
The behavior described in the comment is no longer true after #2269. Argument/return types with jkind any now consistently produce an error.